src/NodeApi/Interop/JSSynchronizationContext.cs

```diff
@@ -42,6 +43,148 @@ public virtual void Dispose()
     public abstract void OpenAsyncScope();
 
     public abstract void CloseAsyncScope();
+
+    public void Post(Action action, bool allowSync = false)
```

Comment:
>Post [](http://example.com/codeflow?start=16&length=4)

I am finding the function that behaves quite differently depending on the bool parameter to be much more difficult to understand and work with. It is better to keep unique name for a function that can run code synchronously. E.g. we use term InvokeElsePost in Office code.

Reply:
This method was moved with same semantics from `JSRuntimeContext`. But we can change it.